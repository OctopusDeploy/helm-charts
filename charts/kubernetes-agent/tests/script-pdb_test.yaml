suite: "podDisruptionBudget"
templates:
  - templates/script-pdb.yaml
tests:
  - it: "is not created when script PDB is disabled"
    set:
      scriptPodDisruptionBudgetEnabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: "is created when script PDB is enabled"
    set:
      scriptPodDisruptionBudgetEnabled: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: PodDisruptionBudget
      - isAPIVersion:
          of: policy/v1
      - equal:
          path: metadata.name
          value: script-pod-disruption-budget
      - equal: 
          path: spec.maxUnavailable
          value: 0
      - equal:
          path: spec.selector.matchExpressions[0].key
          value: octopus.com/scriptTicketId
      - equal: 
          path: spec.selector.matchExpressions[0].operator
          value: Exists