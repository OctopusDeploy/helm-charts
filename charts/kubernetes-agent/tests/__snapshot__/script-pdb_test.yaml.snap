should match snapshot:
  1: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: script-pod-disruption-budget
      namespace: NAMESPACE
    spec:
      maxUnavailable: 0
      selector:
        matchExpressions:
          - key: octopus.com/scriptTicketId
            operator: Exists
